src/main/java/uk/gov/hmcts/reform/sandl.snlevents/mappers/FactsMapper.java

```diff
@@ -47,24 +48,38 @@ public FactsMapper() {
         objectMapper.registerModule(simpleModule);
     }
 
-    public String mapCreateSessionToRuleJsonMessage(CreateSession createSession) throws JsonProcessingException {
+    public String mapCreateSessionToRuleJsonMessage(UpsertSession upsertSession) throws JsonProcessingException {
         FactSession factSession = new FactSession();
 
-        factSession.setId(createSession.getId().toString());
-        factSession.setDuration(createSession.getDuration());
-        factSession.setStart(createSession.getStart());
-        factSession.setCaseType(createSession.getCaseType());
-        if (createSession.getPersonId() != null) {
-            factSession.setJudgeId(createSession.getPersonId().toString());
+        factSession.setId(upsertSession.getId().toString());
+        factSession.setDuration(upsertSession.getDuration());
+        factSession.setStart(upsertSession.getStart());
+        factSession.setCaseType(upsertSession.getCaseType());
+        if (upsertSession.getPersonId() != null) {
+            factSession.setJudgeId(upsertSession.getPersonId().toString());
         }
 
-        if (createSession.getRoomId() != null) {
-            factSession.setRoomId(createSession.getRoomId().toString());
+        if (upsertSession.getRoomId() != null) {
+            factSession.setRoomId(upsertSession.getRoomId().toString());
         }
 
         return objectMapper.writeValueAsString(factSession);
     }
 
+    public String mapUpdateSessionToRuleJsonMessage(Session session) throws JsonProcessingException {
+        FactSession factSession = new FactSession();
+
+        factSession.setId(session.getId().toString());
+        factSession.setDuration(session.getDuration());
+        factSession.setStart(session.getStart());
+        factSession.setCaseType(session.getCaseType());
+
+        Optional.ofNullable(session.getRoom()).ifPresent(r -> factSession.setRoomId(r.getId().toString()));
```

Comment:
If optional here please change in the above method too

Reply:
This is a nice approach 👍  and as Remi wrote, can be used in other places in this class.

btw I'm starting to think about adding a method that would return us 'id' as a string in one call instead of doing getId().toString() all the time, but that's just a thought not to change it now.